Runtime/RestrictiveEnvironment/DefaultStepLockHandling.cs

```diff
@@ -65,6 +65,29 @@ public override void Lock(IStepData data, IEnumerable<LockablePropertyData> manu
             }
         }
 
+        private IStepData GetNextStep(ITransition completedTransition)
+        {
+            if (completedTransition.Data.TargetStep != null)
+            {
+                return completedTransition.Data.TargetStep.Data;
+            }
+
+            if (CourseRunner.IsRunning)
+            {
+                ICourseData course = CourseRunner.Current.Data;
+                // Test all chapters, but the last.
+                for (int i = 0; i < course.Chapters.Count - 1; i++)
```

Comment:
```suggestion
return course.Chapters.SkipWhile(chapter => chapter != course.Current)
    .Skip(1)
    .FirstOrDefault()?.FirstStep.Data;`
```

Reply:
Looks like a discussion about readability vs what is shorter 🤔